packages/@react-spectrum/avatar/test/Avatar.test.js

```diff
@@ -0,0 +1,61 @@
+import {Avatar} from '../';
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+describe('Avatar', () => {
+  it('renders an avatar image', () => {
+    render(<Avatar src="http://localhost/some_image.png" />);
+    expect(screen.getByRole('img')).toBeInTheDocument();
+    expect(screen.getByRole('img')).toHaveAttribute('src', 'http://localhost/some_image.png');
+  });
+
+  it('can render an avatar image with an alt', () => {
+    render(<Avatar src="http://localhost/some_image.png" alt="Test avatar" />);
+    expect(screen.getByAltText(/test avatar/i)).toBeInTheDocument();
+  });
+
+  it('can receive focus and show a focus ring', () => {
+    render(<Avatar src="http://localhost/some_image.png" />);
+    userEvent.tab();
+    expect(screen.getByRole('img')).toHaveFocus();
+    expect(screen.getByRole('img')).toHaveAttribute('class', expect.stringMatching(/focus/i));
+  });
+
+  it('supports custom sizes', () => {
+    render(<Avatar src="http://localhost/some_image.png" size="80px" />);
+    expect(screen.getByRole('img')).toHaveStyle({
+      height: '80px',
+      width: '80px'
+    });
+  });
+
+  it('supports custom class names', () => {
+    render(<Avatar src="http://localhost/some_image.png" UNSAFE_className="my-class" />);
+    expect(screen.getByRole('img')).toHaveAttribute('class', expect.stringContaining('my-class'));
+  });
+
+  it('supports style props', () => {
+    render(<Avatar src="http://localhost/some_image.png" isHidden />);
+    expect(screen.getByRole('img', {hidden: true})).toBeInTheDocument();
+  });
+
+  it('supports custom DOM props', () => {
+    render(<Avatar src="http://localhost/some_image.png" data-testid="Test avatar" />);
+    expect(screen.getByTestId(/test avatar/i)).toBeInTheDocument();
+  });
+
+  describe('when isDisabled = true', () => {
```

Comment:
nit: Would be nice to test that press event don't fire/fire depending on if it is disabled or not. Technically handled in the aria tests though heh so not a must do